### DISCUSSION TITLE
20240627 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#100] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

PR Review:

- https://github.com/torrust/torrust-tracker/pull/873

PRs:

I decided to apply some changes to @da2ce7's PR to deeply understand [his changes](https://github.com/torrust/torrust-tracker/pull/873). I also reorganized the UDP mod a little bit.

- https://github.com/torrust/torrust-tracker/pull/913

New issues:

- https://github.com/torrust/torrust-tracker/issues/918
- https://github.com/torrust/torrust-tracker/issues/917

New discussion:

- https://github.com/torrust/torrust-tracker/discussions/920

Reply discussion:

- https://github.com/torrust/torrust-tracker/discussions/908

### Torrust Index

PR Review:

- https://github.com/torrust/torrust-index/pull/624

### Torrust Demo

- Deploy the demo, updating config files to the newer versions.
- https://github.com/torrust/torrust-demo/issues/7

### Operative: Eng. Manager

Prepare one-on-ones with @cgbosse 

### Learning

I've finished reading the book https://rust-for-rustaceans.com/. I was reviewing @da2ce7's [PR](https://github.com/torrust/torrust-tracker/pull/873), and I did not have any idea about the low-level asynchronous staff like the [Future](https://doc.rust-lang.org/std/future/trait.Future.html) trait.

**Comment by mario-nt** [comment]
### Index
	

- **https://github.com/torrust/torrust-index/pull/624**

> - [x] Load model from string
> - [x] Load policy from string
> - [x] More concrete panic error messages
> - [x] More readable model definitions
> - [x] Test Casbin logging options
> - [x] More readable policy file
> - [x] Code cleanup and refactor


### Training/Research


- Casbin docs 
- Rust programming language book
- Features flags in Rust packages
